### PR TITLE
Update Swin MIM output class

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
- Replaces incorrectly named `logits` output of `SwinMaskedImageModelingOutput` and `SwinV2MaskedImageModelingOutput` classes with the `reconstruction` attribute
- Sets `logits` as a property for backward compatibility and adds a deprecation warning

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for updating!
